app/cluster/[clusterId]/page.tsx

```diff
@@ -37,19 +50,90 @@ export default async function ClusterDetailsPage({
 }: ClusterDetailsPageProps) {
   const { clusterId } = await params
 
-  let cluster: Cluster | undefined
+  let cluster: Awaited<ReturnType<typeof getCluster>>
   try {
-    cluster = (await getCluster(clusterId)) as Cluster
+    cluster = await getCluster(clusterId)
     if (!cluster) throw new Error()
   } catch {
     return notFound()
```

Comment:
this is the part that Im not getting...shouldn't this be the other way around?

- throw an error if the promise fails (catch)
- notFound if `cluster` is null